lib/pbench/agent/results.py

```diff
@@ -406,8 +406,13 @@ def __init__(
         if server:
             path = config.get("results", "rest_endpoint")
             uri = f"{server}/{path}"
+            self.ca = None
         else:
             uri = config.get("results", "server_rest_url")
+
+            # If the "server_ca" config variable isn't defined, we expect to verify
+            # using a registered CA.
+            self.ca = config.get("results", "server_ca", fallback=None)
```

Comment:
Is this code up-to-date with our current thinking?

As the code now stands (AFAICT), if we specify `server`, then we ignore the `server_ca` option in the config file, and we'll depend on `REQUESTS_CA_BUNDLE` if an override is needed; if, instead, we pull the `uri` from the config file, then we'll honor `server_ca` if it is provided...why the bifurcation?  Why not use the config file CA value in both cases if it is provided?  Or, why have the `server_ca` option at all when the user can use `REQUESTS_CA_BUNDLE`?

Reply:
The config file defines a server and optionally a CA to go with that server. If you don't create a custom config file and use `--server`, there's not going to be a `server_ca` anyway. But if you have a config file, even with a `server_ca`, we can't necessarily assume that the `--server` target cert uses that CA.

So basically, config file server goes with config file CA; while `--server` goes with `REQUESTS_CA_BUNDLE`.

Reply:
I'm unclear why we need this change and why it's better than just relying on `REQUESTS_CA_BUNDLE` for both cases.  🤷

Reply:
So your idea is that if someone installs the Agent RPM and relies on a server configured in `pbench-agent.cfg` so they can just `pbench-results-move` (e.g., the passthrough server) we should still require them to put a customized `REQUESTS_CA_BUNDLE` in the environment?

Granted, in the long run, we don't want to force users to rely on private CA certs at all, but this still seems worth doing.

Reply:
My thinking is that, in production, access to the Pbench Server will require only publicly-available and trusted CAs (for some definition of "public"...e.g., "Red Hat internal", perhaps...), and so no specific action (e.g., defining `REQUESTS_CA_BUNDLE`) on the part of the user would be required.

For Staging, we should be operating in a similar environment (i.e., behind the IntLab proxies and using IntLab-signed or RH-IT-signed certs).

So, it's only for func-test and development that we would need to be using our private CA.  "Normal users" wouldn't need it, unless they are working with bits that are too hot for Staging.

And, I think we have func-test already covered.  So, it all really boils down to how do we support our developers.  And, for that, I think we can rely on defining `REQUESTS_CA_BUNDLE` (and, we can probably do it implicitly inside a wrapper script or two, so it doesn't have to be reflected at all in the actual Agent or Server code).